app/javascript/tests/public_health/actions/UpdateCaseStatus.test.js

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import UpdateCaseStatus from '../../../components/public_health/actions/UpdateCaseStatus';
+import { mockMonitoringReasons } from '../../mocks/mockMonitoringReasons';
+import { mockPatient1, mockPatient2, mockPatient3, mockPatient4, mockPatient5 } from '../../mocks/mockPatients';
+
+const onCloseMock = jest.fn();
+const authyToken = "Q1z4yZXLdN+tZod6dBSIlMbZ3yWAUFdY44U06QWffEP76nx1WGMHIz8rYxEUZsl9sspS3ePF2ZNmSue8wFpJGg==";
+
+function simulateStateChange (patientsArray) {
+  // UpdateCaseStatus makes an asynchronous call to obtain values about the selectedPatients
+  // Since we have the full patient objects here, simulate that call by running the same code
+  // as the callback here. This sets the State properly so the component behaves as expected
+  const distinctCaseStatus = [...new Set(patientsArray.map(x=>x.case_status))];
+  const distinctIsolation = [...new Set(patientsArray.map(x=>x.isolation))];
+  const distinctMonitoring = [...new Set(patientsArray.map(x=>x.monitoring))];
+
+  const state_updates = {};
+  if (distinctCaseStatus.length === 1 && distinctCaseStatus[0] !== null) {
+    state_updates.initialCaseStatus = distinctCaseStatus[0];
+    state_updates.case_status = distinctCaseStatus[0];
+  }
+  if (distinctIsolation.length === 1 && distinctIsolation[0] !== null) {
+    state_updates.initialIsolation = distinctIsolation[0];
+    state_updates.isolation = distinctIsolation[0];
+  }
+  if (distinctMonitoring.length === 1 && distinctMonitoring[0] !== null) {
+    state_updates.initialMonitoring = distinctMonitoring[0];
+    state_updates.monitoring = distinctMonitoring[0];
+  }
+  return state_updates
+}
+
+function getMountedWrapper(patientsArray) {
+  let instance = mount(<UpdateCaseStatus
+    authenticity_token={authyToken}
+    patients={patientsArray}
+    monitoring_reasons={mockMonitoringReasons}
+    close={onCloseMock}
+    />);
+
+  let state_updates = simulateStateChange(patientsArray)
+  if (Object.keys(state_updates).length) {
+    instance.setState(state_updates)
+    // no need to use a callback here as the state changes
+    // appear to be performed serially
+  }
+  return instance
+}
+
+const CASE_STATUS_OPTIONS = ['', 'Confirmed', 'Probable', 'Suspect', 'Unknown', 'Not a Case']
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('UpdateCaseStatus', () => {
+  it('Properly renders all main components for Patients who are in Exposure only', () => {
+    // mockPatien2 and mockPatient3 are both in the Exposure workflow
+    const mountedWrapper = getMountedWrapper([mockPatient2, mockPatient5]);
```

Comment:
You shouldn't need to mount in any of these tests since there are no child components, please change to shallow